src/first_pass.c

```diff
@@ -42,77 +39,147 @@ boolean analyze_input_line(char* asm_line)
 	if (opcode != -1)/*if it is stop operation*/ // In second pass refactor i remove opcode global from the function
 		return operation(asm_line + 4);
 	if (is_label(asm_line))
-		return label_actions(asm_line);
+		return label_actions(asm_line, macro_map);
 
 	error_log("line %d: The command was not found\n", line_counter);
 	return FALSE;
 }
 
- /*Address method*/
-char addressing_mode(char* li)
-{
+/**
+ * @brief Checks and sets the addressing mode for an immediate address.
+ *
+ * This function checks if the given operand is a valid immediate address (prefixed with `#`).
+ * It validates that the characters following the `#` are digits, or optionally a `+` or `-` sign,
+ * followed by digits. If the operand is valid, the addressing mode is set to `'0'`.
+ *
+ * @param li A pointer to the operand string.
+ * @param addressing_mode A pointer to a character where the addressing mode will be stored.
+ * @return A boolean value indicating whether the immediate address is valid.
+ *         Returns TRUE if valid, FALSE otherwise.
+ */
+boolean immediate_address(char* li, char* addressing_mode) {
 	int i = 2;
-	li = delete_first_spaces(li);
-	if (*li == '#')
+	boolean result = TRUE;
+	/*If after # there is not a number - throw an error*/
+	if (li[1] < 47 || li[1]>58)
 	{
-		/*If after # does not appear a number - throw an error*/
-		if (li[1] < 47 || li[1]>58)
+		if (li[1] != '-' && li[1] != '+')
 		{
-			if (li[1] != '-' && li[1] != '+')
-			{
-				printf("ERROR!! line %d: Invalid parameter for the instant address\n", line_counter);
-				error_flag = ON;
-			}
+			error_log("line %d: Invalid parameter for the instant address\n", line_counter);
+			result = FALSE;
 		}
-		else
+	}
+	else
+	{
+		while (li[i] != ' ' && li[i] != ',' && li[i] != '\0' && li[i] != '\n' && li[i] != '\t')
```

Comment:
You can use the constant definitions
Also for clean code only if you want - create a variable curr_char and use it instead if li[I]
A batter option is to use a for loop like this
for (current_chat = li, <you full conditiopn>, current_char++)